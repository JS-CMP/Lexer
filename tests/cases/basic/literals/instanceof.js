var func = function () {}

if (func instanceof Function) {
    console.log("func is a function")
}