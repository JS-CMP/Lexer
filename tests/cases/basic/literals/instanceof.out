#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {
    JS::Any func;
    func = JS::Any(JS::InternalObject::create<JS::Function>([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {

        {
        }
        return JS::Any();
    }, 0, u"Anonymous"));
    if (func instanceof Function) 
    {
        console[u"log"](JS::Any(u"func is a function"));
    }
    return 0;
}
